File: Arrays & Hashing/Java/topKFrequentElements.java
import java.util.HashMap;
import java.util.PriorityQueue;
import java.util.Map;

class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> numToFreq = new HashMap<>();
        for (int num : nums) {
            if (numToFreq.containsKey(num)) {
                numToFreq.computeIfPresent(num, (key, value) -> value+1);
            }
            else {
                numToFreq.put(num, 1);
            }
        }

        PriorityQueue<Map.Entry<Integer, Integer>> heap = new PriorityQueue<>
                (numToFreq.size(), (pair1, pair2) ->
                        pair2.getValue() - pair1.getValue());
        heap.addAll(numToFreq.entrySet());

        int[] topKInt = new int[k];
        for (int i = 0; i < k; i++) {
            topKInt[i] = heap.poll().getKey();
        }
        return topKInt;
    }
}

File: Arrays & Hashing/python/top_k_frequent_elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        num_to_freq = {}
        most_freq = []
        for num in nums:
            num_to_freq.setdefault(num, 0)
            num_to_freq[num] += 1
        for i in range(k): 
            most_freq_num = max(num_to_freq, key=num_to_freq.get)
            freq = num_to_freq.pop(most_freq_num)
            most_freq.append(most_freq_num)
        return most_freq
